src/test/java/BaseTest.java

```diff
@@ -1,11 +1,98 @@
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.Parameters;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.time.Duration;
 
 public class BaseTest {
 
+    public static WebDriver driver = null;
+    public static WebDriverWait wait = null;
+    public static String url = null;
+    public static Actions actions = null;
+
+    //"https://qa.koel.app/";
+
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+
+    @BeforeMethod
+    @Parameters({"BaseURL"})
+    public static void launchBrowser(String baseURL) throws MalformedURLException {
+
+        driver = pickBrowser(System.getProperty("browser"));
+        wait = new WebDriverWait(driver,Duration.ofSeconds(10));
+        actions = new Actions(driver);
+        url = baseURL;
+        navigateToPage();
+
+    }
+    public static void navigateToPage() {
+        driver.get(url);
+    }
+
+    @AfterMethod
+    public void closeBrowser() {
+        driver.quit();
+    }
+
+    public static WebDriver pickBrowser(String browser) throws MalformedURLException {
+        DesiredCapabilities caps = new DesiredCapabilities();
+        String gridURL = "http://10.0.0.43:4444";
+
+        switch(browser)
+        {
+            case "MicrosoftEdge": //gradle clean test -Dbrowser=MicrosoftEdge
+                WebDriverManager.edgedriver().setup();
+                EdgeOptions edgeOptions = new EdgeOptions();
+                edgeOptions.addArguments("--remote-allow-origins=*");
+                edgeOptions.addArguments("--disable-notifications");
+                edgeOptions.addArguments("--incognito");
+                edgeOptions.addArguments("--start-maximized");
+                edgeOptions.setExperimentalOption("excludeSwitches", new String[]{"enable-automation"});
+                return driver = new EdgeDriver(edgeOptions);
+
+            case "grid-edge": //gradle clean test -Dbrowser=grid-edge
+                caps.setCapability("browserName","MicrosoftEdge");
+                return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), caps);
+
+            case "grid-chrome": //gradle clean test -Dbrowser=grid-chrome
+                caps.setCapability("browserName","chrome");
+                return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), caps);
+
+            default:
+                WebDriverManager.chromedriver().setup();
+                ChromeOptions chromeOptions = new ChromeOptions();
+                chromeOptions.addArguments("--remote-allow-origins=*");
+                chromeOptions.addArguments("--disable-notifications");
+                chromeOptions.addArguments("--incognito");
+                chromeOptions.addArguments("--start-maximized");
+                chromeOptions.setExperimentalOption("excludeSwitches", new String[]{"enable-automation"});
+                return driver = new ChromeDriver(chromeOptions);
```

Comment:
Hello,

I have checked your homework 24, and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Alex, aliaksandr@testpro.io)!

![804deb64dc406a20a75da05d91a1a69e](https://user-images.githubusercontent.com/104761482/235961479-9afb2841-b657-46c9-99f8-2402f933b843.jpg)